app/src/main/java/br/com/useblu/oceands/client/ui/carouselwithcomponents/CarouselWithComponentsActivity.kt

```diff
@@ -0,0 +1,103 @@
+package br.com.useblu.oceands.client.ui.carouselwithcomponents
+
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.ViewModelProvider
+import br.com.useblu.oceands.client.R
+import br.com.useblu.oceands.client.databinding.ActivityCarouselBinding
+import br.com.useblu.oceands.client.databinding.ActivityCarouselWithComponentsBinding
+import br.com.useblu.oceands.components.compose.OceanCarousel
+import br.com.useblu.oceands.components.compose.OceanCarouselWithComponents
+import br.com.useblu.oceands.components.compose.OceanTheme
+import br.com.useblu.oceands.components.compose.content.OceanCardGroup
+import br.com.useblu.oceands.model.OceanCarouselItem
+import br.com.useblu.oceands.model.compose.carouselwithcomponents.OceanCarouselComponentItem
+import br.com.useblu.oceands.model.compose.carouselwithcomponents.OceanCarouselCycle
+import br.com.useblu.oceands.model.compose.carouselwithcomponents.OceanCarouselIndicator
+
+class CarouselWithComponentsActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityCarouselWithComponentsBinding
+    private lateinit var viewModel: CarouselViewWithComponentsModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_carousel_with_components)
+        binding.lifecycleOwner = this
```

Comment:
```suggestion
class CarouselWithComponentsActivity : AppCompatActivity() {

    private lateinit var viewModel: CarouselViewWithComponentsModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
```

pelo que vi tu nao ta usando o layout xml, entao pode remover as chamdas pra inflar ele, e tbm deletar o arquivo do xml do layout

Reply:
Feito!